### PR TITLE
feat: introduce StatusLoadingIndicator component

### LINKED ISSUE #2360: Component Library: introduce loading indicator component
Some components need a way to show a loading indicator animation. We already have `LoadingImage` for that https://github.com/status-im/status-desktop/blob/master/ui/shared/LoadingImage.qml#L4.

It does the job quite well, we should move it into the component library, ensure that the loading SVG comes with it (and isn't located in `ui/app/img` and we probably should rename it to `StatusLoadingIndicator` or something like that.

It uses `SVGImage` under the hood, which is really just an `Image` that sets the source width and height, so we can probably just inline that: https://github.com/status-im/status-desktop/blob/master/ui/shared/SVGImage.qmlma:

Figma: https://www.figma.com/file/MtAO3a7HnEH5xjCDVNilS7/%F0%9F%8E%A8-Design-System-%E2%8E%9C-Desktop?node-id=706%3A1566

Although, we already have the asset and everything, just need to move the `LoadingImage` to the componen library and maybe rename it etc

### PR DESCRIPTION
**Pull request by PascalPrecht**
A `StatusIcon` that rotates infinitely and can be used for indicating
pending states.

Usage:

```
StatusLoadingIndicator {
    width: 24 // default: 17
    height: 24 // default: 17
    color: "red" // default: loading asset color
}
```

This also removes `LoadingImage` component from `ui/shared`.

Closes #2360

### CODE REVIEW

**Review by jrainville — APPROVED** [review]

File: ui/shared/status/StatusLoadingIndicator.qml [hunk 1]
```diff
@@ -1,14 +1,13 @@
 import QtQuick 2.13
 import "."
 
-SVGImage {
-    id: loadingImg
-    source: "../app/img/loading.svg"
-    width: 17
+StatusIcon {
+    id: root
```
Comment (borismelnik) [inline ui/shared/status/StatusLoadingIndicator.qml:5]: i know than all qml examples use `id: root`, but for many library components inner id is the same as file or i use `id: __impl`. Double underscore is old hack for telling than its a local id.

File: ui/shared/status/StatusLoadingIndicator.qml [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+    icon: "loading"
     height: 17
-    fillMode: Image.Stretch
+    width: 17
```
Comment (borismelnik) [inline ui/shared/status/StatusLoadingIndicator.qml:8]: Maybe we can use `implicitWidth`? Why i mention it:
`StatusLoadingIndicator {
         width: someSituation ? 15 : implicitWidth
         ....`
 What do you think? its make sense?

File: ui/shared/status/StatusLoadingIndicator.qml (hunk 1 above)
Comment (PascalPrecht) [reply ui/shared/status/StatusLoadingIndicator.qml:5]: > but for many library components inner id is the same as file

That's interesting! I don't have strong opinions how we name the root of a component. However we should stay consistent, which means, we'll have to touch many existing components that by now use `root`.

File: ui/shared/status/StatusLoadingIndicator.qml (hunk 2 above)
Comment (PascalPrecht) [reply ui/shared/status/StatusLoadingIndicator.qml:8]: Unfortunately, that doesn't work when component's `implicitWidth` and `implicitHeight` properties are read-only, which is the case of this component: https://github.com/status-im/status-desktop/pull/2435#discussion_r620994414

**Review by borismelnik — APPROVED** [review]

**Pull request merged**